frontend/src/utils/DropDownAutoCompleteTextArea.svelte

```diff
@@ -0,0 +1,89 @@
+<style>
+  .dropdown {
+    text-align: left;
+    position: absolute;
+    margin-top: 3px;
+  }
+
+  textarea {
+    background: inherit;
+    color: inherit;
+    border: none;
+    border-bottom: 1px solid white;
+    flex-grow: 1;
+    height: 10em;
+    margin-left: 1ch;
+  }
+
+  textarea:focus {
+    outline: none;
+    box-shadow: inset 0 -1px 0 var(--main-fg-color);
+  }
+</style>
+
+<script>
+  import { onMount } from "svelte";
+  import DropDown from "./DropDown.svelte";
+
+  export let comment, options, pattern;
+  let autoOptions = options;
+  let dropdown, string, input;
+
+  document.onkeyup = handle;
+
+  function updateInput() {
+    let match = comment.match(pattern);
+    comment = comment.replace(match[0], string);
+  }
+
+  function hideMenu() {
+    dropdown.style.display = "none";
+  }
+
+  function showMenu(e) {
+    dropdown.style.display = "block";
+    const c = document.createElement("canvas");
+    const ctx = c.getContext("2d");
+    ctx.font = `${getComputedStyle(e.target).getPropertyValue(
+      "font-size"
+    )} ${getComputedStyle(e.target).getPropertyValue("font-family")}`;
+    let textSize = ctx.measureText(comment);
```

Comment:
Probably worth refactoring this into `helpers.js` or some other file where it can live standalone.